library/Businessprocess/BpNode.php

```diff
@@ -241,7 +241,7 @@ public function isMissing()
                 }
             }
             $bp->endLoopDetection($this->name);
-            $this->missing = ! $exists && ! empty($this->getChildren());
+            $this->missing = (! $exists && ! empty($this->getChildren())) || $bp->isFaulty();
```

Comment:
It think it's enough to return early if the config is faulty. A faulty config doesn't have any children, so the foreach and loop detection is of no use.